Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -574,10 +568,36 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
 {
-    return (NSInteger)[self.threadMappings numberOfItemsInSection:(NSUInteger)section];
+    NSInteger result = (NSInteger)[self.threadMappings numberOfItemsInSection:(NSUInteger)section];
+    if (self.homeViewMode == HomeViewMode_Inbox) {
+        // Add the "archived conversations" row.
+        result++;
```

Comment:
I chose to use a new row in the same section for the "archived conversations" button.